scripts/autotask.js

```diff
@@ -0,0 +1,1901 @@
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'to',
+        type: 'address',
+      },
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256',
+      },
+    ],
+    name: 'transfer',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool',
+      },
+    ],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'from',
+        type: 'address',
+      },
+      {
+        internalType: 'address',
+        name: 'to',
+        type: 'address',
+      },
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256',
+      },
+    ],
+    name: 'transferFrom',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool',
+      },
+    ],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'unpause',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [
+      {
+        internalType: 'uint256',
+        name: 'amount',
+        type: 'uint256',
+      },
+    ],
+    name: 'withdraw',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+];
+
+// const BPNORI_ADDRESS = `0x8D7713C76A2A6Ec517FB3C032524dd86425E24f4`;
+
+const BPNORI_ADDRESS = "0xBD16B6880F76CA8A3543dd7A2a60d375d0485dBB";
+const MARKET_ADDRESS = `0x53D76F0d411F8Fc5278999Ae7929C163Dbeac7d7`;
+
+const RELAYER_WALLET = `0x6dc772f80495f47d8000530a59ee975b67b7c646`;
+
+const eip712Domain = {
+  Permit: [
+    { name: 'owner', type: 'address' },
+    { name: 'spender', type: 'address' },
+    { name: 'value', type: 'uint256' },
+    { name: 'nonce', type: 'uint256' },
+    { name: 'deadline', type: 'uint256' },
+  ],
+};
+
+exports.handler = async function (event) {
+  console.log(JSON.stringify(event, null, 2));
+  const { amount, buyerWalletAddress, graphqlEndpoint, apiSecret } =
+    event.request.body;
+  if (apiSecret !== event.secrets.apiSecret) return;
+  const provider = new DefenderRelayProvider(event);
+  const signer = new DefenderRelaySigner(event, provider);
+  const store = new KeyValueStoreClient(event);
+  const bpNoriContract = new ethers.Contract(
+    BPNORI_ADDRESS,
+    BPNORI_ABI,
+    signer
+  );
+  const marketContract = new ethers.Contract(
+    MARKET_ADDRESS,
+    MARKET_ABI,
+    signer
+  );
+  const owner = await signer.getAddress();
+  const chainId = 80001;
+  const name = await bpNoriContract.name();
+  const nonce = await bpNoriContract.nonces(owner);
+  const latestBlockTime = (await provider.getBlock('latest')).timestamp;
+  // const deadline = latestBlockTime + 3600; // 1 hour into future
+  const deadline = ethers.constants.MaxUint256;
```

Comment:
why was this needed?

Reply:
originally i'd set the permit deadline for one hour after the current block time. switched it to intmax as part of debugging just in case that was the problem and i was doing my math wrong or something.  I think it's TBD what we want the official deadline to be, and actually, something that Richie and i identified as an issue is the fact that if this relayer is handling a lot of cash transactions in parallel or near parallel, there's a race condition where the relayer will create more than one permit that use the same nonce.  so our actual solution here will potentially involve specifying the market as an approved operator for all of the relayer's NORI to handle cash transactions, or creating a large, long-lived permit from the relayer for the market to spend its funds